content/chainguard/chainguard-images/debugging-distroless-images.md

```diff
@@ -0,0 +1,143 @@
+
+
+## 2. Using Ephemeral (or Debug) Containers
+
+Suppose you have a Kubernetes cluster and one of the containers is having issues. Perhaps it doesn't seem to be able to connect to other services in the cluster, or expected processes aren't running. If the container has a shell, you'd be able to use `kubectl exec` to run debugging commands from inside the container to help diagnose the issue. With a minimal image, this isn't possible, but we can achieve something very similar using `kubectl debug` to launch an ephemeral container.
+
+Let's see an example. We'll start by running a Chainguard nginx image on a Kubernetes cluster:
+
+```
+kubectl run nginx --image=cgr.dev/chainguard/nginx:latest
+```
+
+Which should result in:
+
+```
+pod/nginx created
+```
+
+We can try to start a shell in this pod:
+
+```
+kubectl exec -it nginx -- sh
+```
+
+But the following error occurs:
+
+```
+error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "67bd5164394b1170ac1846bd77ea0b826332365970fbde3b3201bb9abec9b72c": OCI runtime exec failed: exec failed: unable to start container process: exec: "sh": executable file not found in $PATH: unknown
+```
+
+Which is a long way of telling us there is no shell in the image. What we can do instead is use an ephemeral container. An ephemeral container will connect to the namespaces of an existing container, effectively allowing you to sideload debugging tools. Let's try that now:
+
+```
+kubectl debug -it nginx --image=cgr.dev/chainguard/wolfi-base --target=nginx
+```
+
+You should get output similar to:
+
+```
+Targeting container "nginx". If you don't see processes from this container it may be because the container runtime doesn't support this feature.
+Defaulting debug container name to debugger-87792.
+If you don't see a command prompt, try pressing enter.
+nginx:/#
+```
+
+Now we can try running some debugging commands. Let's start by seeing what's running:
+
+```
+nginx:/# ps aux
+PID   USER     TIME  COMMAND
+    1 65532     0:00 nginx: master process /usr/sbin/nginx -c /etc/nginx/nginx.conf -e /dev/stderr
+   10 65532     0:00 nginx: worker process
+   11 65532     0:00 nginx: worker process
+   12 65532     0:00 nginx: worker process
+   13 65532     0:00 nginx: worker process
+   14 65532     0:00 nginx: worker process
+   15 65532     0:00 nginx: worker process
+   16 65532     0:00 nginx: worker process
+   23 root      0:00 /bin/sh -l
+   32 root      0:00 ps aux
+```
+
+And the open ports:
+
+```
+nginx:/# netstat -lntu
+Active Internet connections (only servers)
+Proto Recv-Q Send-Q Local Address           Foreign Address         State
+tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN
+```
+
+The filesystem of the other container may not be accessible due to default permissions.
+
```

Comment:
I think we need something to wrap this up. Maybe this can point to Kubernetes debugging or something at least. It feels a little unfinished in terms of what the dev can achieve here and what their next step would be if there seem to be no problems